### PR TITLE
Feature 2254 roles page selection

### PR DESCRIPTION
**Pull request by mvolkmann**

### CODE REVIEW

File: web-ui/src/components/admin/roles/Roles.jsx [hunk 1]
```diff
@@ -62,10 +62,30 @@ const Roles = () => {
   memberProfiles?.sort((a, b) => a.name.localeCompare(b.name));
 
   useEffect(() => {
-    if (roles) {
```
Comment (mvolkmann) [inline web-ui/src/components/admin/roles/Roles.jsx:65]: I don't think we want to get roles from state now. I think we only want to get them from a query parameter.

File: web-ui/src/components/admin/roles/Roles.jsx [hunk 2]
```diff
@@ -62,10 +62,30 @@ const Roles = () => {
...
-      setSelectedRoles(roles.map(roleObj => roleObj.role));
-    }
-  }, [roles]);
+    const url = new URL(location.href);
```
Comment (mvolkmann) [inline web-ui/src/components/admin/roles/Roles.jsx:65]: This gets the current searchText and selectedRoles to use from query parameters.

File: web-ui/src/components/admin/roles/Roles.jsx [hunk 3]
```diff
@@ -62,10 +62,30 @@ const Roles = () => {
...
+    setSearchText(url.searchParams.get('searchText') ?? '');
+    const selectedRoles = url.searchParams.get('selectedRoles');
+    const roles = selectedRoles?.split(',') ?? [];
+    if (roles.length === 0) roles.push('ADMIN', 'MEMBER', 'PDL');
```
Comment (mvolkmann) [inline web-ui/src/components/admin/roles/Roles.jsx:69]: I don't think we ever want to start with no roles selected. So if none are present in the query parameter then we make them all selected.

File: web-ui/src/components/admin/roles/Roles.jsx [hunk 4]
```diff
@@ -62,10 +62,30 @@ const Roles = () => {
...
+    setSelectedRoles(roles);
+  }, []);
```
Comment (mvolkmann) [inline web-ui/src/components/admin/roles/Roles.jsx:71]: This `useEffect` only runs each time the page is loaded.

File: web-ui/src/components/admin/roles/Roles.jsx [hunk 5]
```diff
@@ -62,10 +62,30 @@ const Roles = () => {
...
+
+  useEffect(() => {
```
Comment (mvolkmann) [inline web-ui/src/components/admin/roles/Roles.jsx:73]: This runs every time the user changes the searchText or selectedRoles.
It updates the query parameters in the URL.

File: web-ui/src/components/admin/roles/__snapshots__/Roles.test.jsx.snap [hunk 6]
```diff
@@ -75,7 +75,7 @@ exports[`renders correctly 1`] = `
               <p
                 class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
               >
-                Showing 2/2 roles
+                Showing 3/2 roles
```
Comment (mvolkmann) [inline web-ui/src/components/admin/roles/__snapshots__/Roles.test.jsx.snap:78]: "3/2" seems wrong. Maybe I need to dig into this test more.

**Review by vhscom — APPROVED** [review]

**Review by S78901 — APPROVED** [review]

**Pull request merged**